server/src-lib/Hasura/RQL/DDL/Schema/Function.hs

```diff
@@ -89,7 +89,22 @@ mkFunctionInfo qf systemDefined config rawFuncInfo =
       when (retTyTyp /= PGKindComposite) $ throwValidateError FunctionReturnNotCompositeType
       unless retSet $ throwValidateError FunctionReturnNotSetof
       unless returnsTab $ throwValidateError FunctionReturnNotSetofTable
-      when (funTy == FTVOLATILE) $ throwValidateError FunctionVolatile
+      -- We mostly take the user at their word here and will, e.g. expose a
+      -- function as a query if it is marked VOLATILE (since perhaps the user
+      -- is using the function to do some logging, say). But this is also a
+      -- footgun we'll need to try to document (since `VOLATILE` is default
+      -- when volatility is omitted). See the original approach here:
+      -- https://github.com/hasura/graphql-engine/pull/5858 
+      --
+      -- This is the one exception where we do some validation. We're not
+      -- commited to this check, and it would be backwards compatible to remove
+      -- it, but this seemed like an obvious case:
```

Comment:
Do you think this documentation has been implemented adequately?

Perhaps, in order to avoid footgunning, can we output some kind of warning message when users expose a `VOLATILE` function as a `query` field?

Reply:
Yeah I decided to punt on; I was on the fence anyway, and there wasn't a `logger` available in nearby callers